### PR TITLE
[Llama3] Change prefill padding in LlamaGenerator to nearest 2048 and optimize chunked prefill readback

### PR DESCRIPTION
**Pull request by skhorasganiTT**
### Ticket
N/A

### Problem description
- The prefill qkv matmul in llama attention assumed that the seq len was divisible by 2048 but seq lens were being padded to the nearest 1024
- Chunked prefill was reading back logits for every chunk instead of only the last one (reducing perf)

### What's changed
- Modified `llama_common.py::get_padded_prefill_len` and `llama_common.py::get_max_prefill_chunk_size` to determine padded prefill len / chunk size based on multiples of 2048 instead of 1024
- Removed `llama_vision_model.py::_get_padded_prefill_seqlen` since it's now a duplicate of `llama_common.py::get_padded_prefill_len`
- Added `MAX_QKV_MM_SEQ_LEN` to model config, replacing hard-coded values of 2048 in `llama_attention.py::prefill_forward`
- Added assert in `llama_attention.py::prefill_forward` before qkv matmul checking divisibility
- Modified chunked prefill to read back logits only for last chunk

### Checklist
- [x] Post commit CI passes
- [x] Blackhole Post commit (if applicable)
- [x] Model regression CI testing passes (if applicable)
- [x] Device performance regression CI testing passes (if applicable)
- [x] **(For models and ops writers)** Full [new models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) tests passes
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by skhorasganiTT** [comment]
T3K unit/freq/demo llama tests: https://github.com/tenstorrent/tt-metal/actions/runs/12640831036
TG unit/freq llama tests: https://github.com/tenstorrent/tt-metal/actions/runs/12640836011
Single-card demo tests: https://github.com/tenstorrent/tt-metal/actions/runs/12640838682
Post-commit tests: https://github.com/tenstorrent/tt-metal/actions/runs/12640842835

**Review by mtairum — APPROVED** [review]
LGTM

**Review by cglagovichTT — APPROVED** [review]
